firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -243,7 +251,7 @@ private final void doRegistration() {
 
       // FID needs to be registered
       if (persistedInstallationEntry.isUnregistered()) {
-        registerAndSaveFid(persistedInstallationEntry);
```

Comment:
This is regarding the throwing of FirebaseInstallationException above in line 244.
That code is an anti-pattern, because the Exception is thrown and caught by you at the end of the method.
You should never throw an exception that you plan to catch in the same library, because that is misusing exceptions for control flow (i.e. as GOTO statements).
Happy to discuss more in person.

Reply:
This can definitely be improved. I agree with you. To contain the scope, I ll send the changes in another PR.